### PR TITLE
upower: update to 0.99.17

### PR DESCRIPTION
**Pull request by oreo639**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by paper42** [comment]
@cinerea0 does this look fine?

**Comment by cinerea0** [comment]
I'm currently testing this PR's changes and everything is working fine for me.

**Comment by cinerea0** [comment]
@oreo639 Actually, could you change the homepage to https://gitlab.freedesktop.org/upower/upower? The homepage in the template right now points to deprecated tarballs.

File: srcpkgs/upower/template [hunk 1]
```diff
@@ -1,39 +1,35 @@
...
  libimobiledevice-devel polkit-devel"
 checkdepends="python3-dbus python3-packaging"
 short_desc="Abstraction for enumerating power devices"
 maintainer="cinerea0 <cinerea0@protonmail.com>"
 license="GPL-2.0-or-later"
-homepage="https://upower.freedesktop.org"
-distfiles="https://gitlab.freedesktop.org/upower/upower/-/archive/UPOWER_${_distver}/upower-UPOWER_${_distver}.tar.bz2"
-checksum=3393944ce8124a888923d580d2867e6257e0209f7534cc3df2448ecae47d9f8d
+homepage="https://gitlab.freedesktop.org/upower/upower"
+distfiles="https://gitlab.freedesktop.org/upower/upower/-/archive/v${version}/upower-v${version}.tar.bz2"
```
Comment (paper42) [inline srcpkgs/upower/template:20]:
```suggestion
homepage="https://gitlab.freedesktop.org/upower/upower"
changelog="https://gitlab.freedesktop.org/upower/upower/-/raw/master/NEWS"
distfiles="https://gitlab.freedesktop.org/upower/upower/-/archive/v${version}/upower-v${version}.tar.bz2"
```

File: srcpkgs/upower/template [hunk 2]
```diff
@@ -1,39 +1,35 @@
...
 build_helper="gir"
-configure_args="--disable-static --with-backend=linux $(vopt_enable gir introspection)"
+build_style=meson
+configure_args="-Dos_backend=linux -Dsystemdsystemunitdir=no
+ -Dintrospection=$(vopt_if gir enabled disabled) $(vopt_bool gtk_doc gtk-doc)"
 conf_files="/etc/UPower/UPower.conf"
-hostmakedepends="automake gettext-devel glib-devel gobject-introspection gtk-doc
- intltool libtool libxslt pkg-config which"
+hostmakedepends="automake gettext gettext-devel glib-devel gobject-introspection
+ $(vopt_if gtk_doc gtk-doc) libtool docbook-xsl libxslt pkg-config which"
```
Comment (paper42) [inline srcpkgs/upower/template:12]:
```suggestion
hostmakedepends="gettext glib-devel gobject-introspection
 $(vopt_if gtk_doc gtk-doc) docbook-xsl libxslt pkg-config which"
```

**Review by paper42 — COMMENTED** [review]
I don't think this is correct:
```diff
@@ -1,9 +1,9 @@
 /etc/UPower/UPower.conf
 /usr/bin/upower
 /usr/lib/girepository-1.0/UPowerGlib-1.0.typelib
-/usr/lib/udev/rules.d/95-upower-hid.rules
-/usr/lib/udev/rules.d/95-upower-wup.rules
 /usr/libexec/upowerd
+/usr/rules.d/95-upower-hid.rules
+/usr/rules.d/95-upower-wup.rules
 /usr/share/dbus-1/interfaces/org.freedesktop.UPower.Device.xml
 /usr/share/dbus-1/interfaces/org.freedesktop.UPower.KbdBacklight.xml
 /usr/share/dbus-1/interfaces/org.freedesktop.UPower.Wakeups.xml
```

**Comment by oreo639** [comment]
> I don't think this is correct:
> 
> ```diff
> @@ -1,9 +1,9 @@
>  /etc/UPower/UPower.conf
>  /usr/bin/upower
>  /usr/lib/girepository-1.0/UPowerGlib-1.0.typelib
> -/usr/lib/udev/rules.d/95-upower-hid.rules
> -/usr/lib/udev/rules.d/95-upower-wup.rules
>  /usr/libexec/upowerd
> +/usr/rules.d/95-upower-hid.rules
> +/usr/rules.d/95-upower-wup.rules
>  /usr/share/dbus-1/interfaces/org.freedesktop.UPower.Device.xml
>  /usr/share/dbus-1/interfaces/org.freedesktop.UPower.KbdBacklight.xml
>  /usr/share/dbus-1/interfaces/org.freedesktop.UPower.Wakeups.xml
> ```

Thanks for letting me know.
eudev has `udevdir`, but upower is looking for `udev_dir`. Not sure how best that should be fixed, but I just added a sed command to replace it for now.

**Comment by oreo639** [comment]
I also tried patching udev_dir into upower and that worked as well, I sent it upstream:
https://github.com/eudev-project/eudev/pull/225

Just so you know for the future.

**Comment by paper42** [comment]
> I also tried patching udev_dir into eudev and that worked as well, I sent the patch upstream: [eudev-project/eudev#225](https://github.com/eudev-project/eudev/pull/225)

Thanks! I think we could backport that commit to our eudev to make sure other packages don't hit this issue and that we don't forget to remove the vsed when eudev is updated.

**Pull request merged**